ERCS/erc-7498.md

```diff
@@ -0,0 +1,310 @@
+- trait redemptions: improving the burn-to-redeem experience with [ERC-7496](./eip-7496.md) Dynamic Traits.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+The token MUST have the following interface and MUST return `true` for [ERC-165](./eip-165.md) supportsInterface for `0x1ac61e13`, the 4 byte interfaceId of the below.
+
+```solidity
+interface IERC7498 {
+  /* Events */
+  event CampaignUpdated(uint256 indexed campaignId, Campaign campaign, string metadataURI);
+  event Redemption(uint256 indexed campaignId, uint256 requirementsIndex, bytes32 redemptionHash, uint256[] considerationTokenIds, uint256[] traitRedemptionTokenIds, address redeemedBy);
+
+  /* Structs */
+  struct Campaign {
+    CampaignParams params;
+    CampaignRequirements[] requirements; // one requirement must be fully satisfied for a successful redemption
+  }
+  struct CampaignParams {
+    uint32 startTime;
+    uint32 endTime;
+    uint32 maxCampaignRedemptions;
+    address manager; // the address that can modify the campaign
+    address signer; // null address means no EIP-712 signature required
+  }
+  struct CampaignRequirements {
+    OfferItem[] offer;
+    ConsiderationItem[] consideration;
+    TraitRedemption[] traitRedemptions;
+  }
+  struct TraitRedemption {
+    uint8 substandard;
+    address token;
+    bytes32 traitKey;
+    bytes32 traitValue;
+    bytes32 substandardValue;
+  }
+
+  /* Getters */
+  function getCampaign(uint256 campaignId) external view returns (Campaign memory campaign, string memory metadataURI, uint256 totalRedemptions);
+
+  /* Setters */
+  function createCampaign(Campaign calldata campaign, string calldata metadataURI) external returns (uint256 campaignId);
+  function updateCampaign(uint256 campaignId, Campaign calldata campaign, string calldata metadataURI) external;
+  function redeem(uint256[] calldata considerationTokenIds, address recipient, bytes calldata extraData) external payable;
+}
+
+---
+
+/* Seaport structs, for reference, used in offer/consideration above */
+enum ItemType {
+    NATIVE,
+    ERC20,
+    ERC721,
+    ERC1155
+}
+struct OfferItem {
+    ItemType itemType;
+    address token;
+    uint256 identifierOrCriteria;
+    uint256 startAmount;
+    uint256 endAmount;
+}
+struct ConsiderationItem extends OfferItem {
+    address payable recipient;
+    // (note: psuedocode above, as of this writing can't extend structs in solidity)
+}
+struct SpentItem {
+    ItemType itemType;
+    address token;
+    uint256 identifier;
+    uint256 amount;
+}
+```
+
+### Creating campaigns
+
+When creating a new campaign, `createCampaign` MUST be used and MUST return the newly created `campaignId` along with the `CampaignUpdated` event. The `campaignId` MUST be an incrementing counter starting at `1`.
```

Comment:
It might just be me, but I find:

> MUST be an incrementing counter starting at `1`

To be a little ambiguous. Perhaps:

> The `campaignId` MUST be an counter incremented with each new campaign. The first campaign MUST have an id of `1`.

Reply:
updated in https://github.com/ethereum/ERCs/pull/27/commits/6cf49638075f962bf1c5196611d167a13e7e3e95, thanks!